deployer/gateway_fqdn_deployer.go

```
@@ -96,17 +94,9 @@ func (k *GatewayFQDNDeployer) Cancel(ctx context.Context, gw *workloads.GatewayF
 
 	oldDeployments := k.tfPluginClient.StateLoader.currentNodeDeployment
 
-	// construct new deployments to have all old deployments except the given one
-	newDeployments := make(map[uint32]gridtypes.Deployment)
-	for nodeID := range oldDeployments {
-		if gw.NodeID != nodeID {
-			newDeployments[nodeID] = gridtypes.Deployment{}
-		}
-	}
-
-	currentDeployments, err := k.deployer.Cancel(ctx, oldDeployments, newDeployments)
-	gw.ContractID = currentDeployments[gw.NodeID]
-	if gw.ContractID == 0 {
+	err = k.deployer.Cancel(ctx, oldDeployments[gw.NodeID])
+	if err == nil {
```

we can return the error  first if exists